sphinx/domains/math.py

```diff
@@ -134,11 +134,14 @@ def get_objects(self) -> Iterable[tuple[str, str, str, str, str, int]]:
         return []
 
     def has_equations(self, docname: str | None = None) -> bool:
-        if docname:
-            return self.data['has_equations'].get(docname, False)
-        else:
+        if not docname:
             return any(self.data['has_equations'].values())
 
+        return (
+            self.data['has_equations'].get(docname, False)
+            or any(self.has_equations(d) for d in self.env.toctree_includes.get(docname, ()))
```

Comment:
Would save line length, but feel free to reject.
```suggestion
            or any(map(self.has_equations, self.env.toctree_includes.get(docname, ()))
```